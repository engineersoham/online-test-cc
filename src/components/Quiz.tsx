import { ArrowBack, ArrowForward } from '@mui/icons-material';
import { Button, Checkbox, FormControl, FormControlLabel, Grid, Radio, RadioGroup, TextField, Typography } from '@mui/material';
import React, { useEffect, useState } from 'react';
import { Navigate, useLocation, useNavigate } from 'react-router-dom';
import { queEng, queHindi } from '../question';


const Quiz = () => {
    
    const [data, setData] :any = useState();
    const [input, setInput] = useState('');
    const [curPage, setPage] = useState(0);
    const navigate = useNavigate();
    const language = useLocation().state;

    const passData = ()=>{
        if(language === 'english'){
            setData(queEng)
        }else{
            setData(queHindi)
        };
    }

    const redBg = {
        backgroundColor:'red',
        margin:'20px',
        color:'white'
    }

    const grayBg = {
        backgroundColor:'gray',
        margin:'20px',
        color:'white'
    }

    const prev = ()=>{
        setPage((prev:any)=>prev-1)
    }

    const next = ()=>{
        setPage((prev:any)=>prev+1)
    }

    const result = ()=>{
        navigate('/result',{state:data})
    }

useEffect(()=>{
    console.log(language);
    passData()
    console.log(data);
    
},[data]);

const handelClick = (e:any)=>{
    data[curPage].isAnsd = true
    console.log(e.target.value);
    setInput(e.target.value)
    data[curPage].userAns = e.target.value;
    console.log(data[curPage]);
}

const handelArr = (e:any)=>{
    console.log(e.target.value);
    data[curPage].isAnsd = true;
    if(e.target.value !== undefined){
        if(data[curPage].userAns.length < 2){
            data[curPage].userAns.push(e.target.value)
        }else{
            data[curPage].userAns.splice(0,1);
            data[curPage].userAns.push(e.target.value)
        }
    }
    console.log(data[curPage]);

};

const option = ()=>{
    switch(data[curPage]?.type){
        case 'mcq':
            return(
                <>
               <FormControl>
               <RadioGroup 
                defaultValue={data[curPage]?.userAns}
                onClick={handelClick}
                >
                    {data[curPage]?.opt.map((item:any,idx:any)=>{
                        return(
                            <>
                            <FormControlLabel
                            data-testid={`mcq-${idx}`}
                            control={<Radio
                            // checked={item === item.userAns}
                            />}
                            label={item}
                            value={item}
                            />
                            </>
                        )
                    })}
                </RadioGroup>
               </FormControl>
                </>
            );

            case 'fill':
                return(
                    <>
                   <FormControl>
                   <TextField
                   data-testid='fill'
                   value={data[curPage]?.userAns}
                   onChange={handelClick}
                   placeholder='text your answer here'
                   />
                   </FormControl>
                    </>
                );

                case 'bool':
                    return(
                        <>
                       <FormControl>
                       <RadioGroup 
                        defaultValue={data[curPage]?.userAns}
                        onClick={handelClick}
                        >
                            {data[curPage]?.opt.map((item:any,idx:any)=>{
                                return(
                                    <>
                                    <FormControlLabel
                                    data-testid={`bool-${idx}`}
                                    control={<Radio
                                    // checked={item === item.userAns}
                                    />}
                                    label={item}
                                    value={item}
                                    />
                                    </>
                                )
                            })}
                        </RadioGroup>
                       </FormControl>
                        </>
                    );

                    case 'match':
                        return(
                            <>
                           <FormControl>
                               <Typography>
                                   {data[curPage]?.match.map((item:any,idx:any)=>{
                                       return(
                                           <>
                                           <Typography
                                           key={idx}
                                           variant='h6'
                                           >
                                               {item}
                                           </Typography>
                                           </>
                                       )
                                   })}
                               </Typography>
                           <RadioGroup 
                           sx={{mt:3}}
                            defaultValue={data[curPage]?.userAns}
                            onClick={handelClick}
                            >
                                {data[curPage]?.opt.map((item:any,idx:any)=>{
                                    return(
                                        <>
                                        <FormControlLabel
                                        data-testid={`match-${idx}`}
                                        control={<Radio
                                        // checked={item === item.userAns}
                                        />}
                                        label={item}
                                        value={item}
                                        />
                                        </>
                                    )
                                })}
                            </RadioGroup>
                           </FormControl>
                            </>
                        );

                        case 'multi':
                            return(
                                <>
                               <FormControl>
                               <RadioGroup 
                                // defaultValue={data[curPage]?.userAns}
                                onClick={handelArr}
                                >
                                    {data[curPage]?.opt.map((item:any,idx:any)=>{
                                        return(
                                            <>
                                            <FormControlLabel
                                            data-testid={`multi-${idx}`}
                                            control={<Checkbox
                                            // checked={data[cuPage]?.ans.includes(item)}
                                            />}
                                            label={item}
                                            value={item}
                                            />
                                            </>
                                        )
                                    })}
                                </RadioGroup>
                               </FormControl>
                                </>
                            );

            default:
                break;
    }
}

  return (
    <div>
       {data?
       <>
        <Grid sx={{textAlign:'center',mb:3}}>
            {data?.map((item:any,idx:any)=>{
                return(
                    <>
                    <Button 
                    data-testid={`btn-${idx}`} 
                    key={idx} 
                    variant='outlined'
                    style={item.isAnsd ? redBg : grayBg}
                    onClick={(e:any)=>setPage(Number(e.target.innerText)-1)}
                    >
                        {item.id}
                    </Button>
                    </>
                )
            })}
        </Grid>
        <Grid sx={{m:2,ml:10}}>
            <Typography variant='h6'>
              Q{data[curPage]?.id}  {data[curPage]?.que}
            </Typography>
        </Grid>
            
            <Grid sx={{ml:12,mt:3}}>
                {option()}
            </Grid>

        <Grid container sx={{mt:5,ml:10}}>
            <Grid xs={1}>
                <Button variant='outlined' data-testid='back' onClick={prev} disabled={data[curPage]?.id <= 1} >
                    <ArrowBack/>
                </Button>
            </Grid>

            <Grid xs={8}></Grid>

            <Grid xs={1}>
                <Button variant='outlined' data-testid='submit' onClick={result} disabled={data[curPage]?.id < data?.length} >
                    SUBMIT
                </Button>
            </Grid>

            <Grid xs={1}>
                <Button variant='outlined' data-testid='frd' onClick={next} disabled={data[curPage]?.id >= data?.length} >
                    <ArrowForward/>
                </Button>
            </Grid>
        </Grid>
       </>
       :
       null}
    </div>
  )
}

export default Quiz