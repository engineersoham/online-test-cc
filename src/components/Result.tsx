import { Grid, Typography } from "@mui/material";
import React, { useEffect, useState } from "react";
import { useLocation } from "react-router-dom";
import { queEng, queHindi } from "../question";

const Result = () => {
  const data: any = useLocation().state;
  let flag = true;
  let score = 0;

  const myStyle = {
    width: "20rem",
    height: "20rem",
    borderRadius: "50%",
    backgroundImage: `conic-gradient(green 0deg, green ${score * 60}deg, red ${
      score * 60
    }deg, red 360deg )`,
  };

  const rightList = Array.from(data).filter((q: any) => {
    return JSON.stringify(q.ans) === JSON.stringify(q.userAns);
  });

  score = rightList?.length;

  useEffect(() => {
    console.log(score);
  }, [score]);

  useEffect(() => {
    console.log(score);
  }, [score]);

  return (
    <div>
      {data && (
        <>
          <Typography
            variant="h4"
            sx={{ textDecoration: "underline", m: 3, textAlign: "center" }}
          >
            RESULT
          </Typography>
          <Grid container sx={{ ml: 3, mr: 3 }}>
            <Grid sx={{ float: "left" }}>
              <Grid
                sx={{
                  width: "25vw",
                  textAlign: "center",
                  backgroundColor: "lightyellow",
                }}
              >
                <Typography sx={{ m: 2 }} variant="h6">
                  Your Score is <b>{score}/6</b>
                </Typography>
                <Typography sx={{ m: 2 }} variant="h6">
                  Percentage is <b>{((score * 100) / 6).toFixed(2)}</b>
                </Typography>
              </Grid>
              <Grid
                sx={{ width: "35vw", backgroundColor: "lightyellow", p: 5 }}
              >
                {Array.from(data)?.map((item: any, idx: any) => {
                  return item.type !== "multi" ? (
                    <div data-testid="data">
                      <Typography sx={{ mb: 2, mt: 3 }}>
                        Q{item.id} {item.que}
                      </Typography>
                      <Typography sx={{ backgroundColor: "lightgreen" }}>
                        correct answer :- {item.ans}
                      </Typography>
                      <Typography sx={{ backgroundColor: "yellow" }}>
                        your answer :- {item.userAns}
                      </Typography>
                    </div>
                  ) : (
                    <div data-testid="data1">
                      <Typography sx={{ mb: 2, mt: 3 }}>
                        Q{item.id} {item.que}
                      </Typography>
                      <Typography sx={{ backgroundColor: "lightgreen" }}>
                        correct answer :- {item.ans[0]} and {item.ans[1]}
                      </Typography>
                      <Typography sx={{ backgroundColor: "yellow" }}>
                        your answer :- {item.userAns[0]} and {item.userAns[1]}
                      </Typography>
                    </div>
                  );
                })}
              </Grid>
            </Grid>
            <Grid sx={{ float: "right", mt: 20, ml: 60 }}>
              <Grid sx={myStyle}></Grid>
            </Grid>
          </Grid>
        </>
      )}
    </div>
  );
};

export default Result;