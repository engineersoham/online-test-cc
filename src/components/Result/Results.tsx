// import { PieChart } from 'react-minimal-pie-chart';
import { Button } from '@mui/material';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { initialValue } from '../../reducer/reducer';


const Results = () => {
  const RIGHTANSWER = useSelector<initialValue, initialValue['rightans']>(state => state.rightans)
  const WRONGANSWER = useSelector<initialValue, initialValue['wrongans']>(state => state.wrongans)
  const scores = useSelector<initialValue, initialValue['score']>(state => state.score)
  const navigate = useNavigate()

  const start = scores * 60

  const mystyle = {
    height: '20rem',
    width: '20rem',
    backgroundColor: "red",
    backgroundImage: `conic-gradient(green 0deg, green ${start}deg,  red ${start}deg, red 360deg)`,
    borderRadius: "50%"

  }

  return (
    <div >
      
      <div style={{ marginLeft: 50, float: 'left' }}>
      <Button variant='contained' sx={{ml:2,mt:2, mb:1}} onClick={()=>{navigate('/enterpage')}}>Retake</Button>
        <h1 style={{ marginBottom: '40px' }}>Your score is <b>{scores}/6</b></h1>

        <h3><u>RIGHT_ANSWERS</u></h3>
      <ol style={{listStyleType: 'none'}}>
       {RIGHTANSWER.map((item:any) => {
          return (
            <li style={{ color: 'green', marginBottom: '10px' }}> <span style={{color:'black', marginRight:'10px'}}>Q{item.ID}. {item.QUE} </span> <br /> &rarr;  "{item.ANS}"  </li>
          )
        })}
       </ol>

        <h3 style={{ marginTop: '40px' }}><u>WRONG_ANSWERS</u></h3>
        <ol style={{listStyleType: 'none'}}>
        {WRONGANSWER.map((item:any) => {
          return (
            <li style={{ color: 'red', marginBottom: '10px' }}><span style={{color:'black', marginRight:'10px'}}>Q{item.ID}. {item.QUE}  </span><br /> &rarr; "{item.ANS}"  </li>
          )
        })}
        </ol>
      </div>

      <div style={{ float: 'right', marginRight: '10rem', marginTop: '5rem' }}>
      <div title='peichart' style={mystyle}></div>

        {/* <PieChart
          data={[
            { title: 'total', value: 6 - scores, color: 'red', totalValue: 6 },
            { title: 'your score', value: scores, color: 'green', totalValue: 6 }
          ]}
          radius={50}
        />; */}

        

      </div>


    </div>
  )
}

export default Results