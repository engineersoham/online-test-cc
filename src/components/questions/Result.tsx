// import { PieChart } from 'react-minimal-pie-chart';
import { useSelector } from 'react-redux';
import { initialValue } from '../../reducer/reducer';



const Result = () => {
  const RIGHTANSWER = useSelector<initialValue, initialValue['rightans']>(state => state.rightans)
  const WRONGANSWER = useSelector<initialValue, initialValue['wrongans']>(state => state.wrongans)
  const scores = useSelector<initialValue, initialValue['score']>(state => state.score)

  const start = scores * 60

  const mystyle = {
    height: '20rem',
    width: '20rem',
    backgroundColor: "red",
    backgroundImage: `conic-gradient(green 0deg, green ${start}deg,  red ${start}deg, red 360deg)`,
    borderRadius: "50%"

  }

  return (
    <div>
      <div style={{ marginLeft: 50, float: 'left' }}>
        <h1 style={{ marginBottom: '40px' }}>Your score is <b>{scores}/6</b></h1>

        <h3><u>RIGHT_ANSWERS</u></h3>
        {RIGHTANSWER.map(item => {
          return (
            <li style={{ color: 'green', marginBottom: '10px' }}>{item}</li>
          )
        })}

        <h3 style={{ marginTop: '40px' }}><u>WRONG_ANSWERS</u></h3>
        {WRONGANSWER.map(item => {
          return (
            <li style={{ color: 'red', marginBottom: '10px' }}>{item}</li>
          )
        })}
      </div>

      <div style={{ float: 'right', marginRight: '10rem', marginTop: '5rem' }}>
      <div style={mystyle}></div>
        {/* <PieChart
          data={[
            { title: 'total', value: 6 - scores, color: 'red', totalValue: 6 },
            { title: 'your score', value: scores, color: 'green', totalValue: 6 }
          ]}
          radius={50}
        />; */}

        

      </div>


    </div>
  )
}

export default Result