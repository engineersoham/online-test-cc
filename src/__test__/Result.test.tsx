import  Quiz from '../components/Quiz' 
import { fireEvent, render, screen } from "@testing-library/react";
import Result from '../components/Result';

jest.mock("react-router-dom",()=>{
    return{
        
        useLocation:()=>{
            return{
                state:'english'
            }
        }
    }
})

describe('test for result componet',()=>{
    test('snap for result',()=>{
        const tree = render(
            <Result/>
        );
        expect(tree).toMatchSnapshot();
        screen.debug()
    })

    test('test for result componet',()=>{
        const mock = jest.fn();
        render(
            <Result/>
        );
        expect(mock).toBeDefined();
;
        const score = screen.getByText('Your Score is');
        expect(score).toBeInTheDocument();
        
        // const data = screen.getByTestId('data')
        // const data1 = screen.getByTestId('data1');
        // expect(data1).toBeInTheDocument()

    }
    )
})