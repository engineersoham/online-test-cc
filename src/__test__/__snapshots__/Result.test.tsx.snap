// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for result componet snap for result 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-11b88os-MuiTypography-root"
        >
          RESULT
        </h4>
        <div
          class="MuiGrid-root MuiGrid-container css-1ebpurq-MuiGrid-root"
        >
          <div
            class="MuiGrid-root css-1becngf-MuiGrid-root"
          >
            <div
              class="MuiGrid-root css-j1jb3a-MuiGrid-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-1a3mx6w-MuiTypography-root"
              >
                Your Score is 
                <b>
                  7
                  /6
                </b>
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-1a3mx6w-MuiTypography-root"
              >
                Percentage is 
                <b>
                  116.67
                </b>
              </h6>
            </div>
            <div
              class="MuiGrid-root css-rbiswf-MuiGrid-root"
            >
              <div
                data-testid="data"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
                >
                  Q
                   
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
                >
                  correct answer :- 
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
                >
                  your answer :- 
                </p>
              </div>
              <div
                data-testid="data"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
                >
                  Q
                   
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
                >
                  correct answer :- 
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
                >
                  your answer :- 
                </p>
              </div>
              <div
                data-testid="data"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
                >
                  Q
                   
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
                >
                  correct answer :- 
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
                >
                  your answer :- 
                </p>
              </div>
              <div
                data-testid="data"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
                >
                  Q
                   
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
                >
                  correct answer :- 
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
                >
                  your answer :- 
                </p>
              </div>
              <div
                data-testid="data"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
                >
                  Q
                   
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
                >
                  correct answer :- 
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
                >
                  your answer :- 
                </p>
              </div>
              <div
                data-testid="data"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
                >
                  Q
                   
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
                >
                  correct answer :- 
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
                >
                  your answer :- 
                </p>
              </div>
              <div
                data-testid="data"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
                >
                  Q
                   
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
                >
                  correct answer :- 
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
                >
                  your answer :- 
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root css-17960as-MuiGrid-root"
          >
            <div
              class="MuiGrid-root css-yg9aw3-MuiGrid-root"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-11b88os-MuiTypography-root"
      >
        RESULT
      </h4>
      <div
        class="MuiGrid-root MuiGrid-container css-1ebpurq-MuiGrid-root"
      >
        <div
          class="MuiGrid-root css-1becngf-MuiGrid-root"
        >
          <div
            class="MuiGrid-root css-j1jb3a-MuiGrid-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-1a3mx6w-MuiTypography-root"
            >
              Your Score is 
              <b>
                7
                /6
              </b>
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-1a3mx6w-MuiTypography-root"
            >
              Percentage is 
              <b>
                116.67
              </b>
            </h6>
          </div>
          <div
            class="MuiGrid-root css-rbiswf-MuiGrid-root"
          >
            <div
              data-testid="data"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
              >
                Q
                 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
              >
                correct answer :- 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
              >
                your answer :- 
              </p>
            </div>
            <div
              data-testid="data"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
              >
                Q
                 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
              >
                correct answer :- 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
              >
                your answer :- 
              </p>
            </div>
            <div
              data-testid="data"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
              >
                Q
                 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
              >
                correct answer :- 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
              >
                your answer :- 
              </p>
            </div>
            <div
              data-testid="data"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
              >
                Q
                 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
              >
                correct answer :- 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
              >
                your answer :- 
              </p>
            </div>
            <div
              data-testid="data"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
              >
                Q
                 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
              >
                correct answer :- 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
              >
                your answer :- 
              </p>
            </div>
            <div
              data-testid="data"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
              >
                Q
                 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
              >
                correct answer :- 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
              >
                your answer :- 
              </p>
            </div>
            <div
              data-testid="data"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-lvs09v-MuiTypography-root"
              >
                Q
                 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-15108kq-MuiTypography-root"
              >
                correct answer :- 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2qxmvk-MuiTypography-root"
              >
                your answer :- 
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root css-17960as-MuiGrid-root"
        >
          <div
            class="MuiGrid-root css-yg9aw3-MuiGrid-root"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
